### DISCUSSION TITLE
Percy build:wait fails in github action

### DISCUSSION THREAD

**Discussion by diogofscmariano** [percy/cli#952] [Support] [answered]
Hi
I've setup a github action with checks components using storybook capabilties, and after it finishes I need to check the result of the checks. For that, I'm trying to use the build:wait. Although it fails with 

`
[percy:client] Waiting for build 1724df30/React-Component-Library (HEAD)... (27ms)
`
`
[percy:client] Fetching builds for 1724df30/React-Component-Library (0ms)
`
```
[percy:cli] TypeError: Cannot read properties of undefined (reading 'attributes')
    at logProgress (file:///home/runner/work/react-component-library/react-component-library/node_modules/@percy/cli-build/dist/wait.js:66:21)
    at file:///home/runner/work/react-component-library/react-component-library/node_modules/@percy/cli-build/dist/wait.js:50:5
    at poll (file:///home/runner/work/react-component-library/react-component-library/node_modules/@percy/client/dist/client.js:221:67)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)(134ms)
```

Not much to debug from. The error does not seem to be from the authentication token. Usually a 403 is fired, at least locally. Same command works locally tho.
`yarn percy build:wait -f -p 1724df30/React-Component-Library -c HEAD -v`

Node: 16.15.0
@percy/cli: 1.2.1

Config file: 
`
{
  version: 2,
  snapshot: {
    widths: [
      1280
    ],
    minHeight: 1024,
    percyCSS: ''
  },
  discovery: {
    networkIdleTimeout: 100
  },
  upload: {
    files: '**/*.{png,jpg,jpeg}',
    ignore: '',
    stripExtensions: false
  }
}
`

Any hint?

**Comment by yjaaidi** [comment]
Hi @diogofscmariano, I just stumbled on the same issue while switching from `percy-poller` to `percy build:wait`.

The issue happens for workflows triggered by `pull_request` because the commit sha of the PR is the merge commit not the last commit.
`percy start` is not using that sha, it is currently using a sha from the PR event payload. Cf. https://github.com/percy/cli/blob/master/packages/env/test/github.test.js

There are two alternatives:
1. run Percy only on `push` events
2. try to get the sha from he PR payload then fallback to GITHUB_SHA

```yaml
run: yarn percy build:wait --project 1724df30/React-Component-Library --commit "${PR_HEAD_SHA:-$GITHUB_SHA}" --fail-on-changes
env:
  PR_HEAD_SHA: ${{ github.event.pull_request.head.sha }}
```

**Comment by wwilsman** [comment, marked as answer]
Hey @diogofscmariano & @yjaaidi!

I wanted to chime in to say we've addressed this issue in this PR (https://github.com/percy/cli/pull/1008) as well as add in an extra feature so you no longer need to handle the `pull_request.head.sha` yourself!

Once that is merged and released, you'll be able to omit the `--commit` flag, causing the command to default to the commit found in the environment. Which, in the case of GitHub Actions, is exactly the event payload & fallback @yjaaidi suggested https://github.com/percy/cli/blob/937e5d081ea610191569bddcbf0bb2731d16f823/packages/env/src/environment.js#L105

**Reply by yjaaidi** [reply]
Cool ! Thanks @wwilsman
